Sources/TuistCore/Graph/GraphTraverser.swift

```diff
@@ -44,6 +44,20 @@ public class GraphTraverser: GraphTraversing {
         })
     }
 
+    public func allInternalTargets() -> Set<GraphTarget> {
+        let allInternalTargets = projects
+            .filter { projectPath, _ in
+                !projectPath.pathString.contains("\(Constants.tuistDirectoryName)/\(Constants.DependenciesDirectory.name)")
+            }
+            .flatMap { projectPath, project -> [GraphTarget] in
+                let targets = graph.targets[projectPath, default: [:]]
+                return targets.values.map { target in
+                    GraphTarget(path: projectPath, target: target, project: project)
+                }
+            }
+        return Set(allInternalTargets)
+    }
+
```

Comment:
Maybe you can simplify this by implementing it as a filter on the `allTargets` method?

Reply:
Maybe it could even be an extension in the `GraphTraversing.swift` so it would apply to all implementations 🤔